iree/compiler/Dialect/VM/Conversion/StandardToVM/ConvertStandardToVM.cpp

```diff
@@ -783,6 +783,30 @@ class CallOpConversion : public OpConversionPattern<CallOp> {
   }
 };
 
+class FloorDivSIOpConversion : public OpConversionPattern<arith::FloorDivSIOp> {
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult matchAndRewrite(
+      arith::FloorDivSIOp srcOp, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    if (adaptor.lhs().getType() != adaptor.rhs().getType())
+      return rewriter.notifyMatchFailure(srcOp, "argument types must match");
+
+    if (adaptor.lhs().getType().isInteger(32)) {
```

Comment:
it'd be worth making this if-else if-else to keep the supported cases easier to handle - then we can return rewriter.notifyMatchFailure instead of just failure to get nice diagnostics (printing "unsupported type " << type etc)